install.sh

```diff
@@ -388,19 +394,31 @@ setup_env() {
         info "Using default agent configuration directory ${CATTLE_AGENT_CONFIG_DIR}"
     fi
 
+    # --- install to /var/lib/rancher/agent by default, except if we are running within transactional-update
+    # --- in which case we install into /etc/rancher/agent/var as /var is not mounted to the snapshot.
     if [ -z "${CATTLE_AGENT_VAR_DIR}" ]; then
-        CATTLE_AGENT_VAR_DIR=/var/lib/rancher/agent
-        info "Using default agent var directory ${CATTLE_AGENT_VAR_DIR}"
+        if check_rootfs_rw && [ -x /usr/sbin/transactional-update ]; then
```

Comment:
Can we nest the check_rootfs_rw call under an if that checks for transactional update? I don’t think it makes sense for us to touch a file in / unless we know we have transactional update. That greatly reduces my regression concerns.

Reply:
I’m not in front of a computer but we may also just be able to reverse the order of the arguments in the existing if, as long as we stop evaluation once a failure occurs. I’m not in front of a computer able to test this atm though.